src/Hazelcast.Net/NearCaching/NearCacheBase.cs

```diff
@@ -73,6 +73,11 @@ protected NearCacheBase(string name, Cluster cluster, SerializationService seria
             _evictionComparer = GetEvictionComparer(_evictionPolicy);
             _evictionPercentage = nearCacheOptions.EvictionPercentage;
             _cleanupInterval = nearCacheOptions.CleanupPeriodSeconds * 1000;
+
+            cluster.State.Failover.ClusterOptionsChanged += (options) =>
+            {
+                 Clear();
```

Comment:
In Java, we clear near caches after we connect & authenticate to the new cluster. It is the same here, right?

Reply:
This will be addressed after the release